module/zfs/vdev_raidz.c

```diff
@@ -2290,6 +2414,22 @@ vdev_raidz_child_done(zio_t *zio)
 	rc->rc_error = zio->io_error;
 	rc->rc_tried = 1;
 	rc->rc_skipped = 0;
+
+	/*
+	 * Process the disk io latency before it goes out of scope.
+	 *
+	 * A zio->io_delay value of zero means this IO was part of
+	 * an aggregation.
+	 * We want columns that are at least 2 sectors high so that
+	 * the map read was across all the data columns.
+	 */
+	uint64_t two_sectors = 2ULL << zio->io_vd->vdev_top->vdev_ashift;
+	if (zio->io_type == ZIO_TYPE_READ && zio->io_error == 0 &&
+	    zio->io_size >= two_sectors && zio->io_delay != 0) {
```

Comment:
Could you explain why do we care about the two sectors (all data columns) here?

Not accounting aggregated ZIOs makes this algorithm even more random that periodic sampling alone would do.  With RAIDZ splitting ZIOs between vdevs into smaller ones, they are good candidates for aggregation.